### PR TITLE
Added code-comments for all the sim functions

### PR DESCRIPTION
**Pull request by alihitawala**
@kvpradap 
Please review.

### CODE REVIEW

File: py_stringmatching/simfunctions.py [hunk 1]
```diff
@@ -63,21 +65,24 @@ def affine(string1, string2, gap_start=1, gap_continuation=0.5, sim_score=sim_id
...
+    # DP initialization
     for j in _range(1, len(string2) + 1):
         m[0][j] = -float("inf")
         x[0][j] = -float("inf")
         y[0][j] = gap_start + (j - 1) * gap_continuation
-
+    # affine gap calculation using DP
     for i in _range(1, len(string1) + 1):
         for j in _range(1, len(string2) + 1):
+            # best score between x_1....x_i and y_1....y_j given that x_i is aligned to y_j
```
Comment (kvpradap) [inline py_stringmatching/simfunctions.py]: __[flake8]__
* [E501] line too long (91 > 79 characters)

File: py_stringmatching/simfunctions.py [hunk 2]
```diff
@@ -682,36 +722,48 @@ def soft_tfidf(bag1, bag2, corpus_list=None, sim_func=jaro, threshold=0.5):
...
         for element in document:
+            # adding element only if it is present in one of two input string
             if element in bag1 or element in bag2:
                 temp_set.add(element)
                 total_unique_elements.add(element)
+        # update element document frequency for this document
         for element in temp_set:
             element_freq[element] = element_freq[element] + 1 if element in element_freq else 1
     similarity_map = {}
+    # calculating the term sim score against the input string 2, construct similarity map
```
Comment (kvpradap) [inline py_stringmatching/simfunctions.py]: __[flake8]__
* [E501] line too long (89 > 79 characters)

File: py_stringmatching/simfunctions.py [hunk 3]
```diff
@@ -682,36 +722,48 @@ def soft_tfidf(bag1, bag2, corpus_list=None, sim_func=jaro, threshold=0.5):
...
     for x in bag1:
         if x not in similarity_map:
             max_score = 0.0
             for y in bag2:
                 score = sim_func(x, y)
+                # adding sim only if it is above threshold and highest for this element
```
Comment (kvpradap) [inline py_stringmatching/simfunctions.py]: __[flake8]__
* [E501] line too long (87 > 79 characters)

**Comment by kvpradap** [comment]
Are you planning to update the following monge elkan examples in this branch?

  >>> monge_elkan(['Comput.', 'Sci.', 'and', 'Eng.', 'Dept.,', 'University', 'of', 'California,', 'San', 'Diego'], ['Department', 'of', 'Computer', 'Science,', 'Univ.', 'Calif.,', 'San', 'Diego'])
        0.8677218614718616
        >>> monge_elkan(['Comput.', 'Sci.', 'and', 'Eng.', 'Dept.,', 'University', 'of', 'California,', 'San', 'Diego'], ['Department', 'of', 'Computer', 'Science,', 'Univ.', 'Calif.,', 'San', 'Diego'], sim_func=needleman_wunsch)
        2.0
        >>> monge_elkan(['Comput.', 'Sci.', 'and', 'Eng.', 'Dept.,', 'University', 'of', 'California,', 'San', 'Diego'], ['Department', 'of', 'Computer', 'Science,', 'Univ.', 'Calif.,', 'San', 'Diego'], sim_func=affine)
        2.25

**Comment by alihitawala** [comment]
Sorry I missed it. I will fix this in this branch. I will notify you once
done.

-Ali

**Comment by alihitawala** [comment]
Hi Pradap,

I am facing one issue, I hope that you would be able to help me with this.
I couldn't run the tests in the code base I cloned from the master. It is
giving following errors. This is the same error I am receiving in PyCharm.

[image: Inline image 1]

Can you please help me fix this.

Thanks,
Ali

**Comment by kvpradap** [comment]
Interesting, I cloned a master from scratch and ran the test cases using nose tests and it ran without any errors. Moreover, Travis CI also did the build and ran the test cases successfully.

Do the following steps
1) Copy you current py_stringmatching directory to a temp directory
2) Clone a fresh copy of py_stringmatching (master branch) package from GitHub
3) Run nosetests from the command line.

If the above fails,

1) Push your changes  
2) Create a pull request and check whether the build goes through and test cases passed by travis CI

File: py_stringmatching/tests/test_simfunctions.py [hunk 4]
```diff
@@ -394,8 +394,8 @@ def test_valid_input(self):
 
         self.assertEqual(monge_elkan(['Niall'], ['Neal']), 0.8049999999999999)
         self.assertEqual(monge_elkan(['Niall'], ['Njall']), 0.88)
-        # self.assertEqual(monge_elkan(['Comput.', 'Sci.', 'and', 'Eng.', 'Dept.,', 'University', 'of', 'California,', 'San', 'Diego'],
-        #                              ['Department', 'of', 'Computer', 'Science,', 'Univ.', 'Calif.,', 'San', 'Diego']), 0.8677218614718616)
+        self.assertEqual(monge_elkan(['Comput.', 'Sci.', 'and', 'Eng.', 'Dept.,', 'University', 'of', 'California,', 'San', 'Diego'],
```
Comment (kvpradap) [inline py_stringmatching/tests/test_simfunctions.py]: __[flake8]__
* [E501] line too long (133 > 79 characters)

File: py_stringmatching/tests/test_simfunctions.py [hunk 5]
```diff
@@ -394,8 +394,8 @@ def test_valid_input(self):
...
+                                     ['Department', 'of', 'Computer', 'Science,', 'Univ.', 'Calif.,', 'San', 'Diego']), 0.8364448051948052)
```
Comment (kvpradap) [inline py_stringmatching/tests/test_simfunctions.py]: __[flake8]__
* [E501] line too long (139 > 79 characters)

**Comment by alihitawala** [comment]
I changed the Python version to 3.5 from 2.7 - then I started getting
errors for py_stringmatching module import error. Finally, after putting
the right value for PYTHONPATH it got resolved.

But I am still not sure why is it giving that error with 2.7. I checked out
clean code from master and ran the tests from CLI, then too I was getting
the same error as before. For now, I would run this in Python3.5.

Let me know if there is something I am missing.

-Ali

**Comment by alihitawala** [comment]
@kvpradap  you can go ahead with the review.

Thanks

**Comment by kvpradap** [comment]
Travis CI is not complaining any error (it would have built the code across 2.7, 3.3, 3.4 and 3.5 and ran nose tests). I merged the pull request and tested again, the tests pass for py: 2.7, 3.3, 3.4 and 3.5.
And the error that you reported does not seem to be arising because of the difference between python 2 and 3. 

This means that the error would be in your local machine and some issue with the configuration. Try this:
1) pip uninstall py_stringmatching
2) Remove or move all the python versions except 2.7 and test again.

If the above doesnt work, try installing and testing ina cslab machine.

**Pull request merged**